main.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'planet'
+require_relative 'solar_system'
+
+def main
+  solar_system = SolarSystem.new('Sol')
+ # hard coding planets into solar_system
+  earth = Planet.new('Earth', 'blue-green', 5.972e24, 1.496e8, 'Only planet known to support life')
+  mercury = Planet.new('Mercury', 'grey', 4.454e90, 3.45e9, 'Mercury fun fact')
+  venus = Planet.new('Venus', 'orange', 6.43e93, 4.12e9, 'Venus fun fact')
+  mars = Planet.new('Mars', 'red', 4.238e82, 2.19e4, 'Mars fun fact')
+  jupiter = Planet.new('Jupiter', 'brown', 6.492e30, 4.19e6, 'Jupiter fun fact')
+ # adding hard-coded planets to system
+  solar_system.add_planet(earth)
+  solar_system.add_planet(mercury)
+  solar_system.add_planet(venus)
+  solar_system.add_planet(mars)
+  solar_system.add_planet(jupiter)
+ # loop for user input
+  loop do
+    puts "Would you like to 1. List planets, \n 2. See planet details, \n 3. Add planet, or \n 4. Exit? Enter a number"
+    input = gets.chomp!
+    if input == "1"
+      list = solar_system.list_planets
+      puts list
+      exit
+    elsif input == "2"
+      puts "Which planet would you like to learn about?"
+      user_input = gets.chomp!
+      found_planet = solar_system.find_planet_by_name(user_input)
+      puts found_planet
```

Comment:
For the above two lines...

ideally, calling `solar_system.find_planet_by_name(user_input)` would return an instance of `Planet`. Therefore, the value of `found_planet` would be an instance of `Planet`, and in the above line, you can then execute `puts found_planet.summary`, so that we use the method `.summary` defined on `Planet`, which will give us the string we need.